packages/react-components/src/utils/researchOutputForm.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  ResearchOutputResponse,
+  ResearchOutputIdentifierType,
+  DecisionOption,
+  ResearchOutputPublishingEntities,
+  TeamResponse,
+} from '@asap-hub/model';
+import { ComponentProps } from 'react';
+import ResearchOutputContributorsCard from '../organisms/ResearchOutputContributorsCard';
+
+export type ResearchOutputState = {
+  title: string;
+  description: string;
+  link?: string;
+  type: string;
+  tags: readonly string[];
+  methods: string[];
+  organisms: string[];
+  environments: string[];
+  subtype?: string;
+  labCatalogNumber?: string;
+  teams: NonNullable<
+    ComponentProps<typeof ResearchOutputContributorsCard>['teams']
+  >;
+  labs: ComponentProps<typeof ResearchOutputContributorsCard>['labs'];
+  authors: ComponentProps<typeof ResearchOutputContributorsCard>['authors'];
+  identifierType: ResearchOutputIdentifierType;
+  identifier?: string;
+};
+
+export type getTeamState = {
+  team: TeamResponse | undefined;
+  publishingEntity: ResearchOutputPublishingEntities;
+  researchOutputData: ResearchOutputResponse | undefined;
+};
+
+const equals = (a: Array<string>, b: Array<string>): boolean =>
+  a.length === b.length && a.every((element, index) => element === b[index]);
+
+export const getTeamsState = ({
+  team,
+  publishingEntity,
+  researchOutputData,
+}: getTeamState) => {
+  if (publishingEntity === 'Working Group') {
+    return [];
+  }
+  return (
+    researchOutputData?.teams.map((element) => ({
+      label: element.displayName,
+      value: element.id,
+      isFixed: true,
+    })) || [
+      {
+        label: team?.displayName || '',
+        value: team?.id || '',
+        isFixed: true,
+      },
+    ]
+  );
+};
+
+export const isDirtyEditMode = (
```

Comment:
I think that this is very repetitive. Could we make a function that takes the initial state of the form (eg empty or the api version of the research output data) it return a function that you can call with the current state and it will compare. I don't think this will need to vary between the forms as